src/com/zone5cloud/retrofit/core/OkHttpClientInterceptor_Authorization.java

```diff
@@ -60,25 +65,23 @@
 	 *  Pass in a delegate if you want to be notified of updates to the auth token,
 	 *  for instance if you wish to save the token to persistent storage.
 	 *
-	 * @param token - the initial authorization token to be used in the Authorization header for all requests which require it. 
-	 * token may be null and can be set later via setToken(), or automatically set with a login request
-	 * @param clientID - Authentication service API key.
-	 * @param clientSecret - Authentication service API secret, Cognito only. Set to null for Gigya keys.
+	 * @param clientConfig (gets and sets token, client key , secret and username
 	 * @param delegates - delegates to receive callbacks whenever the token is updated.
 	 */
-	public OkHttpClientInterceptor_Authorization(AuthToken token, String clientID, String clientSecret, Z5AuthorizationDelegate ...delegates) {
-		this.token.set(token);
-		this.clientID = clientID;
-		this.clientSecret = clientSecret;
-		
+	public OkHttpClientInterceptor_Authorization(ClientConfig clientConfig, Z5AuthorizationDelegate ...delegates) {
+		this.token.set(clientConfig.getToken());
+		this.clientID = clientConfig.getClientID();
+		this.clientSecret = clientConfig.getClientSecret();
+		this.zone5BaseUrl = clientConfig.getZone5BaseUrl();
+
 		for (Z5AuthorizationDelegate delegate: delegates) {
 			this.delegates.add(delegate);
 		}
 	}
-	
+
 	/** Set the Authentication service API key and secret. Set clientSecret to null for Gigya keys */
 	public void setClientIDAndSecret(String clientID, String clientSecret) {
-		this.clientID = clientID;
+		this.clientSecret = clientID;
 		this.clientSecret = clientSecret;
 	}
```

Comment:
revert change. Needs to be `this.clientID = clientID`, not `this.clientSecret = clientID`